src/Bicep.Core/Registry/Providers/TypesV1Archive.cs

```diff
@@ -24,15 +27,20 @@ public static async Task<Stream> GenerateProviderTarStream(IFileSystem fileSyste
             var indexJson = await fileSystem.File.ReadAllTextAsync(indexJsonPath);
             await AddFileToTar(tarWriter, "index.json", indexJson);
 
-            /* TODO:
-               * read index.json
-               * figure out paths to other .json files that need to be included
-               * read other .json files, add them to tgz
-            */
+            var indexJsonParentPath = Path.GetDirectoryName(indexJsonPath);
+            var uniqueTypePaths = getAllUniqueTypePaths(indexJsonPath, fileSystem);
+
+            foreach (var relativePath in uniqueTypePaths)
+            {
+                var absolutePath = Path.Combine(indexJsonParentPath!, relativePath);
+                var typesJson = await fileSystem.File.ReadAllTextAsync(absolutePath);
+                await AddFileToTar(tarWriter, relativePath, typesJson);
+            }
         }
 
-        stream.Seek(0, SeekOrigin.Begin);
-        return stream;
+        //stream.Seek(0, SeekOrigin.Begin);
+
+        return new MemoryStream(stream.ToArray(), 0, stream.ToArray().Length, true);
```

Comment:
>MemoryStream [](http://example.com/codeflow?start=19&length=12)

I think if you remove the `using` from L21, you will be able to return the stream as-is and also avoid an extra array allocation. 

In general, you should only dispose disposable objects only if you're done using them within the method. Resetting position to 0 and returning the stream without disposing is perfectly valid. (The responsibility to dispose the stream then falls to the caller.)